packages/react/src/components/media/players/VideoPlayer.tsx

```diff
@@ -196,29 +198,49 @@ const HtmlVideoPlayer = React.forwardRef<
     muted,
     poster,
     objectFit,
-    onAccessControlError,
+    // onAccessControlError,
     filteredSources,
     fullscreen,
+    onError,
   } = props;
 
-  const handleError = React.useCallback(
-    (error: Error) => {
-      if (isAccessControlError(error)) {
-        onAccessControlError?.(error);
-      }
-      console.warn(error.message);
-    },
-    [onAccessControlError],
-  );
+  const onVideoError: React.ReactEventHandler<HTMLVideoElement> =
+    React.useCallback(
+      async (_e) => {
+        // TODO: find a way to determine what the response code is from the backend
+        // we cannot pull the error code from the HTML video element
+        // and we cannot make a network request to the source URL unless CORS headers are added
+        // this is likely not a good idea, so we need to come up with a different solution
```

Comment:
Hmm what is exactly the error you were getting here? CORS should be certainly fixed. Was it because you were getting that JSON instead of the m3u8 with a proper HLS error? If that, [it is going to get fixed](https://discord.com/channels/423160867534929930/1087726352645488740/1091023592621363280)! We should keep the same error handling logic as before on the player! WDYT?

Reply:
So, that error handling logic never worked for this component. This component is only used for MP4 playback.

The problem is, that MP4 doesn't have CORS headers set, and the video element does not provide an error code.

If we could add CORS headers to the MP4 URL, that would solve this problem.